### PR TITLE
Fix #534: add kickstart http config to rsconf

### LINKED ISSUE #534: add http config for kickstart
No longer serving kickstart files over nfs. Now using http. So, add to http config

```
    location /kickstart/ {
        alias /srv/kickstart/;
    }
```

### PR DESCRIPTION
**Pull request by e-carlin**

### CODE REVIEW

File: rsconf/component/rsconf.py [hunk 1]
```diff
@@ -29,6 +29,10 @@ def internal_build_compile(self):
             auth_f=nginx.CONF_D.join(PASSWD_SECRET_F),
             srv_d=jc.rsconf_db.srv_d,
             host_subdir=jc.rsconf_db.srv_host_d.basename,
+            kickstart_hosts=[
```
Comment (robnagler) [inline rsconf/component/rsconf.py:32]: I don't think this is a good idea. It should just be configuration. This assumes a lot about passwd_secret_f, which it shouldn't Nothing else assumes that except for setup_dev. It also requires the host to be added to rsconf before it has a base operating system. Rather, use a CIDR for the subnet with a list of special cases.

File: tests/pkcli/build1_data/1.out/srv/host/v3.radia.run/etc/nginx/conf.d/v3.radia.run.conf [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        auth_basic_user_file /etc/nginx/conf.d/rsconf_auth;
+        # DEBUG: return 200 $remote_user;
+        if ($remote_user != $1) {
+            return 403;
+        }
+        root /home/vagrant/src/radiasoft/rsconf/tests/pkcli/build1_work/1/srv;
+    }
+
+    location /kickstart/ {
+        allow v9.radia.run;
```
Comment (robnagler) [inline tests/pkcli/build1_data/1.out/srv/host/v3.radia.run/etc/nginx/conf.d/v3.radia.run.conf:35]: This list will get really long with the number of hosts we have. Much better to use a CIDR.

File: rsconf/package_data/rsconf/nginx.conf.jinja [hunk 3]
```diff
@@ -30,4 +30,12 @@ server {
         }
         root {{ rsconf.srv_d }};
     }
+
+    location /kickstart/ {
```
Comment (robnagler) [inline rsconf/package_data/rsconf/nginx.conf.jinja:34]: We don't normally include the trailing slash.

**Review by robnagler — CHANGES_REQUESTED** [review]

File: rsconf/component/rsconf.py (hunk 1 above)
Comment (e-carlin) [reply rsconf/component/rsconf.py]: Use trusted nets or just add config? https://github.com/radiasoft/rsconf/pull/535/commits/c2c36928e70baab657deeb5f723116cd7bcb88a2

File: rsconf/package_data/rsconf/nginx.conf.jinja (hunk 3 above)
Comment (robnagler) [reply rsconf/package_data/rsconf/nginx.conf.jinja:34]: The ones above are regular expressions and we have the trick in dev of having the credentials in /srv so the names would collide with the host dirs.

File: rsconf/component/rsconf.py (hunk 1 above)
Comment (robnagler) [reply rsconf/component/rsconf.py]: Add config. Let the user decide. This is security so using a default would be too much.

File: rsconf/package_data/rsconf/nginx.conf.jinja (hunk 3 above)
Comment (e-carlin) [reply rsconf/package_data/rsconf/nginx.conf.jinja]: 40b6ce1

File: rsconf/component/rsconf.py (hunk 1 above)
Comment (e-carlin) [reply rsconf/component/rsconf.py]: 40b6ce1

**Comment by e-carlin** [comment]
@robnagler ready for another review

**Review by robnagler — APPROVED** [review]

**Pull request merged**